index.js

```diff
@@ -329,18 +316,16 @@ Fetcher.prototype.postProcessFile = function (filePath) {
 Fetcher.prototype.postProcessDownloads = function () {
   var me = this;
 
-  return new Promise(function (resolve, reject) {
-    fs.readdir(me.localPath, function (err, files) {
-      if (err) {
-        logError(err);
-        reject(err);
-        return;
-      }
-      Promise.all(files.map(function (file) {
+  return fsp.readdir(me.localPath)
+    .catch(function (err) {
```

Comment:
yeah ok. 

If i am following the code right, the final `.then` in this promise chain could eventually hit `fsp.writeFile` which throws an error on rejection. It looks like the code will stop there, which is probably what is wanted on a write error